app/src/main/kotlin/com/simplemobiletools/smsmessenger/models/Message.kt

```diff
@@ -27,4 +27,35 @@ data class Message(
     fun isReceivedMessage() = type == Telephony.Sms.MESSAGE_TYPE_INBOX
 
     fun millis() = date * 1000L
+
+    companion object {
+
+        fun getStableId(message: Message): Long {
+            var result = message.id.hashCode()
+            result = 31 * result + message.body.hashCode()
+            result = 32 * result + message.date.hashCode()
+            result = 31 * result + message.threadId.hashCode()
+            result = 31 * result + message.isMMS.hashCode()
+            result = 31 * result + (message.attachment?.hashCode() ?: 0)
+            result = 31 * result + message.senderName.hashCode()
+            result = 31 * result + message.senderPhotoUri.hashCode()
+            result = 31 * result + message.isScheduled.hashCode()
```

Comment:
what is 31 and 32?

Reply:
why use 31: https://stackoverflow.com/a/35304979/8475376

32 was a typo.